applications/vanilla/controllers/class.moderationcontroller.php

```diff
@@ -280,7 +280,7 @@ public function confirmCommentDeletes($discussionID = '') {
             unset($checkedComments[$discussionID]);
             Gdn::userModel()->saveAttribute($session->UserID, 'CheckedComments', $checkedComments);
             ModerationController::informCheckedComments($this);
-            $this->setRedirectTo('discussions');
+            $this->setRedirectTo('discussion/'.$discussionID);
```

Comment:
If this is what you want to do then you want to call the `discussionUrl()` function which takes a full discussion. Fortunately, it looks like you have one above in the `$discussion` variable above.